xbmc/cores/VideoPlayer/DVDSubtitles/SubtitlesAdapter.cpp

```diff
@@ -27,7 +27,7 @@ CSubtitlesAdapter::~CSubtitlesAdapter()
 bool CSubtitlesAdapter::Initialize()
 {
   m_libass->SetSubtitleType(ADAPTED);
-  return m_libass->CreateTrack();
+  return m_libass->CreateTrack() && m_libass->CreateStyle();
```

Comment:
a heads up because i saw this in a different section: `&&` does short-circuit, so if `CreateTrack()` returns false -> `CreateStyle()` will not get executed. if that‘s not intended please use `&` …. or to make it more obvious:
```c
const bool trackCreated = …
const bool styleCreated = …
return trackCreated && styleCreated;

```

Reply:
CreateStyle depends exclusively on CreateTrack i think go well with `&&`

Reply:
Yeah this is intentional. I'll add logging prior to all early returns both in createtrack and createstyle